chips/nrf52/src/comp.rs

```diff
@@ -0,0 +1,301 @@
+use kernel::common::cells::OptionalCell;
+use kernel::common::registers::{
+    register_bitfields, register_structs, ReadOnly, ReadWrite, WriteOnly,
+};
+use kernel::common::StaticRef;
+use kernel::hil::analog_comparator;
+use kernel::ReturnCode;
+/// Representation of an AC channel on the SAM4L.
+pub struct AcChannel {
+    chan_num: u32,
+}
+
+// Only one channel, coden taken from sam4l implementation
+#[derive(Copy, Clone, Debug)]
+#[repr(u8)]
+enum Channel {
+    AC0 = 0x00,
+}
+
+/// Initialization of an AC channel.
+impl AcChannel {
+    /// Create a new AC channel.
+    ///
+    /// - `channel`: Channel enum representing the channel number
+    const fn new(channel: Channel) -> AcChannel {
+        AcChannel {
+            chan_num: ((channel as u8) & 0x0F) as u32,
```

Comment:
That's how it was in the sam4l implementation, but in retrospect, I don't know why. I updated it. In other places, channel is usize. I don't think it matters though because I think it's unlikely a chip will have more than 255 analog comparators.